core/state.py

```diff
@@ -939,6 +939,23 @@ def __exit__(self, type, value, traceback):
         self.mem.PopCall(True)
 
 
+class ctx_Argv(object):
+    """For $1, $2, $3, etc."""
```

Comment:
I just pushed a change noting that $0 isn't set, which is conventionally the whole match rather than a submatch


And Python does support the equivalent of it

It might be that we need to turn state.dollar0 into a stack ... Let me think if there's another way to implement it.  I guess we could also change GetArgNum() or something



```
>>> p=re.compile(r'\d+')
>>> 

>>> p.sub('yy', 'foo 123')
'foo yy'
>>> 
>>> p.sub(r'-\g<0>-', 'foo 123')
'foo -123-'
```

Reply:
I was thinking about this state.dollar0 issue -- I was hesistant to **couple** the unrelated regex and the `argv[0]` semantic

It's kind of mixing two different things up

I think we might not need a stack -- we can just create a separate variable, and then GetArgNum(0) can just test if there's an "active" regex match?

Just a thought

(BTW `state.GetArg0()` does not need to be changed, that's only for word eval slicing)